docs/0-core-use-cases.md

```diff
@@ -0,0 +1,68 @@
+# Core use-cases
+
+### Add key-binding
+
+This function is responsible for "Adding value A for key X" where key X is
+ represented by "key" and value A by "val". Which causes creating the
+  following keymap: `X-> A`. This use-case is used to create a brand new keymap
+   with the first key-binding. Function returns `result->action` which is
+    the database save function call with the keymap as an argument.
+
+### Append key-binding
+
+This function is responsible for "Adding a value A for key X, where key Y has
+ value
+ B."
+where key X is represented by "key-a", value A by "val-a", key Y by "key-b
+" and value B by "val-b". Which will result in following keymap being
+ remembered :
+ 
+ ```
+ X -> A
+ Y -> B
+ ```
+This use-case is used to append existing keymap by adding a new key-binding to
+ correlated key-value pair. First, function checks if keymap which is to be
+  expanded exists, if not function returns `storage-error`. Otherwise, the
+   function checks if the keymap contains the key to be added, if it does, it
+    returns `action/make-error` with type `entity/append` that points
```

Comment:
"that points location of error and value `key-exist` that points cause of error."

Type of an error and cause of an error.